pkg/sbom/spdx/unmarshal.go

```diff
@@ -135,11 +142,53 @@ func (s *SPDX) unmarshal(spdxDocument *spdx.Document) error {
 		s.SBOM.Applications = append(s.SBOM.Applications, *app)
 	}
 
+	// Fallback for when there are no effective relationships.
+	if len(relationships) == 0 {
+		if err := s.parsePackages(spdxDocument); err != nil {
+			return err
+		}
+	}
+
 	// Keep the original document
 	s.SPDX = spdxDocument
 	return nil
 }
 
+// parsePackages processes the packages and categorizes them into OS packages and application packages.
+// Note that all language-specific packages are treated as a single application.
+func (s *SPDX) parsePackages(spdxDocument *spdx.Document) error {
+	var (
+		osPkgs []ftypes.Package
+		app    ftypes.Application
```

Comment:
if spdx doesn't have application - we think that all packages are from 1 application, i understand correctly?

Reply:
Also perhaps we want to use `initApplication`?